### ISSUE TITLE
Docs: Document upgrade path for no-parameter-properties

### ISSUE THREAD

**Issue by Zarel** [typescript-eslint/typescript-eslint#8078]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

I'm upgrading from an old version of typescript-eslint, and the lack of documented upgrade path for `no-parameter-properties` gave me so much headache.

The CLI error message is simply

```
  1:1  error  Definition for rule '@typescript-eslint/no-parameter-properties' was not found  @typescript-eslint/no-parameter-properties
```

And the URL just 404s:

https://typescript-eslint.io/rules/no-parameter-properties

(Despite being linked to from the v6 announcement.)

The v6 release announcement mentions that it's a deprecated rule that was removed, but doesn't mention anything about upgrade path. I clicked _so_ many links and got a lot of 404s and PRs discussing removing deprecated rules, but not what they'd been replaced with.

This is in stark contrast to no-duplicate-imports, which has a page telling me to use import/no-duplicates instead: https://typescript-eslint.io/rules/no-duplicate-imports/

I had to go through the entire rule list to find the intended upgrade path: https://typescript-eslint.io/rules/parameter-properties

Please add a link to this from https://typescript-eslint.io/rules/no-parameter-properties

### Affected URL(s)

https://typescript-eslint.io/rules/no-parameter-properties

**Comment by Zamiell** [comment]
Thanks for the report, do you wanna try sending a PR for us?

**Comment by Josh-Cena** [comment]
Yes, we should add a page like `no-duplicate-imports` and `camelcase`!

**Comment by bradzacher** [comment]
We only keep `camelcase` and `no-duplicate-imports` around solely because those are core rules - so we keep the docs as a tombstone directing people to their alternately named replacements.

So we don't want to restore it - we should instead update the v6 blog post to explicitly list the replacements for the deleted rules.

**Comment by peanutenthusiast** [comment]
> 
> So we don't want to restore it - we should instead update the v6 blog post to explicitly list the replacements for the deleted rules.

Hi @bradzacher ! I'm working on this and just wanted to gauge which rule, if at all, replaces `no-implicit-any-catch` -- perhaps just `no-implicit-any`? Thanks! 

Expect a WIP PR soon.

**Comment by bradzacher** [comment]
There was no replacement rule - instead it was replaced with the tsconfig option `useUnknownInCatchVariables`
https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables

**Comment by Josh-Cena** [comment]
Even if we don't have a tombstone page, we would need to setup a redirect so people from outdated blog posts won't think they are getting gaslit ^^

**Comment by bradzacher** [comment]
Cc @JoshuaKGoldberg some good feedback/ideas for us for the next major release. We need to be clearer about migration paths and we need to be better about preserving old URLs via redirects.

I hadn't thought of the latter before but redirecting a deleted rule's docs to the major blog post is a really good idea.

**Comment by JoshuaKGoldberg** [comment]
👍 agreed. Filed:

* #8105
* #8106
